lisa/sut_orchestrator/azure/transformers.py

```diff
@@ -269,6 +286,80 @@ def _get_public_ip_address(
         return public_ip_address
 
 
+class DeployTransformer(Transformer):
+    """
+    deploy a node in transformer phase for further operations
+    """
+
+    __resource_group_name = "resource_group_name"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "azure_deploy"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return DeployTransformerSchema
+
+    @property
+    def _output_names(self) -> List[str]:
+        return [self.__resource_group_name]
+
+    def _internal_run(self) -> Dict[str, Any]:
+        platform = _load_platform(self._runbook_builder, self.type_name())
+        runbook: DeployTransformerSchema = self.runbook
+
+        envs = Environments()
+        environment_requirement = EnvironmentSpace()
+        environment_requirement.nodes.append(runbook.requirement)
+        environment = envs.from_requirement(environment_requirement)
+        assert environment
+
+        platform.prepare_environment(environment=environment)
```

Comment:
Confused here: how do these calls interact/intersect with lisa_runner's calls of it?

Reply:
They are not related. The logic of platform is reused here to deploy/delete environments. The transformers are used to prepare environments before runners.

Reply:
I see. We'd better document that